packages/core/src/AttesterSchema.ts

```diff
@@ -0,0 +1,278 @@
+        }
+
+        for (let i = 0; i < this.replDataFields.length; i++) {
+            this.replDataFields[i] = new DataField()
+        }
+
+        this.schema = parseSchema(_schema)
+        this.encodedData = new Array(FIELD_COUNT).fill(BigInt(2) ** BigInt(254))
+
+        this.encodedData = new Array(FIELD_COUNT).fill(BigInt(0))
+
+        this.schemaLookup = {}
+
+        this.buildDataFields()
+    }
+
+    findEntryByName(entry: string): {} {
+        for (let i = 0; i < this.schema.length; i++) {
+            if (this.schema[i]['name'] == entry) {
+                return Object.assign({}, this.schema[i])
+            }
+        }
+        return {}
+    }
+
+    buildDataFields() {
+        for (let i = 0; i < this.schema.length; i++) {
+            const name = this.schema[i]['name']
+            const bits = BigInt(this.schema[i]['type'].slice(4))
+            const updateBy = this.schema[i]['updateBy']
+
+            if (updateBy == 'sum') {
+                if (
+                    !this.sumDataFields[this.curSumDataFieldCount].addOrPass(
+                        name,
+                        bits
+                    )
+                ) {
+                    this.curSumDataFieldCount++
+
+                    if (this.curSumDataFieldCount >= SUM_FIELD_COUNT) {
+                        throw Error('Too many sum fields')
+                    }
+                    const res = this.sumDataFields[
+                        this.curSumDataFieldCount
+                    ].addOrPass(name, bits)
+
+                    if (!res) {
+                        throw Error("Couldn't add " + name + ' sum data field')
+                    }
+                }
+            } else if (updateBy == 'replace') {
+                this.replDataFields[this.curReplDataFieldCount].setMaxFieldSize(
+                    254 - REPL_NONCE_BITS
+                )
+                if (bits > 254 - REPL_NONCE_BITS) {
+                    throw Error('Excessive replacement bits allocation')
+                }
+
+                if (
+                    !this.replDataFields[this.curReplDataFieldCount].addOrPass(
+                        name,
+                        bits
+                    )
+                ) {
+                    if (
+                        this.curReplDataFieldCount + SUM_FIELD_COUNT ==
+                        FIELD_COUNT
+                    ) {
+                        throw Error('Too many replacement and sum fields')
+                    }
+                    this.curReplDataFieldCount++
+
+                    const res = this.replDataFields[
+                        this.curReplDataFieldCount
+                    ].addOrPass(name, bits)
+
+                    if (!res) {
+                        throw Error(
+                            "Couldn't replace " +
+                                name +
+                                ' replacement data field'
+                        )
```

Comment:
this could be tested

Reply:
Reworked code so that this condition will never be met. Now, the it will repeatedly try to find available field indices until there are none left.